### PR TITLE
DOC: add Examples and fix docstring for special.airye

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
fix a part of #7168

#### What does this implement/fix?
add Examples and fix docstring for special.airye()

### CODE REVIEW

**Comment by ilayn** [comment]
You might want to use the OOP style as matplotlib fols recommend its usage

```python
from scipy.special import airye
import matplotlib.pyplot as plt
z = np.linspace(40, 200)
eAi, eAip, eBi, eBip = airye(x)
f, ax = plt.subplots(2, 1, sharex=True)
for ind, data in enumerate([[eAi, eBi, ["eAi", "eBi"]],
                            [eAip, eBip, ["eAip", "eBip"]]]):
    ax[ind].plot(z, data[0], "-r", z, data[1], "-b")
    ax[ind].legend(data[2])
    ax[ind].grid()
```

**Review by person142 — CHANGES_REQUESTED** [review]
The interesting thing about these functions is the “exponentially scaled” bit, so for an example I would want to see something that compares them to the normal Airy functions so that you can see what they are exponentially scaled compared to.

**Comment by pv** [comment]
And maybe some example where the usual non-scaled functions under/overflow whereas the exponentially scaled ones do not, so that the reason for the existence of these functions becomes clearer.

**Comment by AtsushiSakai** [comment]
Thank you guys for reviewing and comments. I updated it based on your feedbacks. PTAL.

**Comment by AtsushiSakai** [comment]
Close and reopen to rerun CI

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -369,6 +370,48 @@ def add_newdoc(name, doc):
...
+    function quickly underflows for large values, whereas the exponentially
+    scaled function does not.
+    
+    >>> f, ax = plt.subplots(2, 1, sharex=True)
+    >>> for ind, data in enumerate([[Ai, eAi, ["Ai", "eAi"]],
+    ...                             [Aip, eAip, ["Aip", "eAip"]]]):
+    ...     ax[ind].plot(z, data[0], "-r", z, data[1], "-b")
+    ...     ax[ind].legend(data[2])
+    ...     ax[ind].grid(True)
+    >>> plt.show()
```
Comment (pv) [inline scipy/special/add_newdocs.py:413]: Probably clearer to show this in code than as plots:
```suggestion
>>> airy(200)
(0.0, 0.0, nan, nan)
>>> airye(200)
(0.07501041684381093, -1.0609012305109042, 0.15003188417418148, 2.1215836725571093)
```

**Comment by ilayn** [comment]
This is quite nice now. Thanks @AtsushiSakai @pv

**Pull request merged**